alibi/utils/lang_model.py

```diff
@@ -0,0 +1,279 @@
+import abc
+import numpy as np
+import tensorflow as tf  # type: ignore
+from typing import List, Optional, Tuple
+
+import transformers  # type: ignore
+from transformers import TFAutoModelForMaskedLM, AutoTokenizer
+
+
+class LanguageModel(abc.ABC):
+    SUBWORD_PREFIX = ''
+
+    def __init__(self, model_path: str):
+        """
+        Initialize the language model.
+
+        Parameters
+        ----------
+        model_path
+            `transformers` package model path.
+        """
+        self.model_path = model_path
+
+        # set model
+        self.model = TFAutoModelForMaskedLM.from_pretrained(model_path)
+        self.caller = tf.function(self.model.call)
+
+        # set tokenizer
+        self.tokenizer = AutoTokenizer.from_pretrained(model_path)
+
+    @abc.abstractmethod
+    def is_subword_prefix(self, token: str) -> bool:
```

Comment:
new doc:

```
        """
        Checks if the given token is a part of the tail of a word. Note that a word can
        be split in multiple tokens (e.g, word = [head tail_1 tail_2 ... tail_k]).
        Each language model has a convention on how to mark a tail token. For example
        DistilbertBaseUncased and BertBaseUncased have the tail tokens prefixed with the special
        set of characters `##`. On the other hand, for RobertaBase only the head is prefixed
        with the special character 'Ġ' and thus we need to check the absence of the prefix to identify
        the tail tokens. We call those special characters SUBWORD_PREFIX. Due to different conventions 
        this method has to be implemented for each language model.

        Parameters
        ----------
        token
            Token to be checked if it is a subword.

        Returns
        -------
        True if the given token is a subword. False otherwise.
        """
```
I think that a subword can be either a `head` or a `tail` token. We are really after the tail tokens which a prefixed or not by some special characters depending on the convention.

Reply:
Looks good!